src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-strategy.ts

```diff
@@ -93,6 +93,16 @@ export default class AmazonPayV2PaymentStrategy implements PaymentStrategy {
 
         this._walletButton.click();
 
+        if (this._isModalFlow(region)) {
+            return new Promise((_, reject) => {
+                window.addEventListener(
+                    'focus',
+                    () => reject(new PaymentMethodCancelledError('Shopper needs to login to Amazonpay to continue')),
+                    { once: true }
```

Comment:
Thanks, i've checking browser supported versions and I don't think this could be a problem. Most browsers support it since few years ago. The only browser that not support this is IE9. You think a validation would be required for this scenario?

Reply:
IE11 doesn't support it either: https://caniuse.com/once-event-listener

and we still support IE11: https://github.com/bigcommerce/checkout-sdk-js#browser-support

we can play it safe by using a named function and removing it on invocation instead:

```
        if (this._isModalFlow(region)) {
            return new Promise((_, reject) => {
                const onFocus = () => {
                    window.removeEventListener('focus', onFocus);
                    reject(new PaymentMethodCancelledError('Shopper needs to login to Amazonpay to continue'));
                };

                window.addEventListener( 'focus', onFocus);
            });
        }
```

Reply:
Made the change, thanks. @leeBigCommerce